src/Rings/mpoly-ideals.jl

```diff
@@ -2249,3 +2278,11 @@ function flag_pluecker_ideal(ring::MPolyRing{<: FieldElem}, dimensions::Vector{I
                    isReduced=true,
                    isGB=true))
 end
+
+# Since most ideals implement `==`, they have to implement the hash function.
+# See issue #4143 for problems entailed. Interestingly, this does not yet fix 
+# the failure of unique! on lists of ideals.
+function hash(I::Ideal, c::UInt)
+  return hash(typeof(I), hash(base_ring(I), c))
+end
```

Comment:
I advocate for this solution now. It is a matter of fact that ideals can not be hashed in general and even for polynomial rings this is probably expensive if not impossible (we need to choose an ordering which might be a dead end for a specific ideal). Yet, the `==` function is vital to how we use ideals and we need to be able to use them in vectors/lists/dictionaries.

Reply:
Is it safe to hash the type? After all two mathematically equal objects may have different types. 
(Thinking e.g. of IdealSheaf, RadicalIdealSheaf, PrimeIdealSheaf etc.)

Reply:
> Is it safe to hash the type? After all two mathematically equal objects may have different types. (Thinking e.g. of IdealSheaf, RadicalIdealSheaf, PrimeIdealSheaf etc.)

It is only safe if you don't implement `==` between this type and other types. In your example, you shouldn't include the type in hashes.

Reply:
Ok, good to know. I was only following the suggestion by @benlorenz . But you're probably right.